### PR TITLE
Build with current tool versions

### PR DESCRIPTION
**Pull request by tobiasge**
As I had problems building dashclock on my machine with only the current tools and Java 8 installed I updated the build system to work with the new versions:

- Update grade
- Update build tools
- Force source and target java to 1.7
- use centralized settings for minSdkVersion and targetSdkVersion
- use centralized settings for repositories and android gradle tools

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,4 +1,24 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
-ext.buildToolsVersion = "21.1.2"
+ext.buildToolsVersion = "23.0.2"
+ext.minSdkVersion = 17
+ext.targetSdkVersion = 21
+ext.compileSdkVersion = 21
 
+buildscript {
+  repositories {
```
Comment (romannurik) [inline build.gradle]: can you update these with the right indentation? (4 spaces)

File: build.gradle [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+    mavenCentral()
+  }
+  dependencies {
+    classpath 'com.android.tools.build:gradle:1.5.0'
+  }
+}
+
+subprojects {
+  repositories {
+    maven { url "https://jitpack.io" }
```
Comment (romannurik) [inline build.gradle]: why jitpack?

**Comment by tobiasge** [comment]
- Updated the indentation like requested
- jitpack is there for the Material Dialogs dependency, see:
https://github.com/afollestad/material-dialogs#gradle-dependency

**Comment by romannurik** [comment]
Thanks! Re: jitpack, we're not using the dialogs library AFAICT, so would be good to just remove the two extra repositories.

**Comment by tobiasge** [comment]
MaterialDialog is used in this file: src/main/java/com/example/dashclock/examplehost/ExampleHostActivity.java

**Comment by romannurik** [comment]
Oh, interesting. We should remove that eventually in favor of the support library. Thanks for the heads up. Will merge ASAP after resolving that.

**Comment by tobiasge** [comment]
In this commit https://github.com/tobiasge/dashclock/commit/ca855a6121ac57dbfc1c05dd08b7d1ad739bd05c I have removed the MaterialDialog dep and used the AlertDialog from appcompat-v7 instead.

**Comment by romannurik** [comment]
Oh wow, thank you! Can you drop that commit into this PR?

**Comment by tobiasge** [comment]
Done.

**Pull request merged**